packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx

```diff
@@ -171,6 +182,17 @@ export const AddNewProposalModal = () => {
                 setRationale={(rationale) => send('SET_RATIONALE', { rationale })}
               />
             )}
+            {state.matches('generalParameters.triggerAndDiscussion') && (
+              <TriggerAndDiscussionStep
+                constants={constants as ProposalConstants}
+                trigger={state.context.trigger}
+                discussionMode={state.context.discussionMode}
+                discussionWhitelist={state.context.discussionWhitelist}
+                setTrigger={(trigger?: ProposalTrigger) => send('SET_TRIGGER', { trigger })}
+                setDiscussionMode={(mode) => send('SET_DISCUSSION_MODE', { mode })}
+                setDiscussionWhitelist={(whitelist) => send('SET_DISCUSSION_WHITELIST', { whitelist })}
```

Comment:
WDYT about using a single object and update function? It may also simplify the machine events and you could re-use the interface used for `useForm()` definition